### ISSUE TITLE
request error: Illegal break statement for 200 response with no body

### ISSUE THREAD

**Issue by PiotrSmietana89** [silkimen/cordova-plugin-advanced-http#260]
I'm working on a cordova app that POSTs information to a server, which by design returns response with empty body. This seems to cause some problems for cordova-plugin-advanced-http, even if response status is 200. Failure callback is called instead of success, with error message:

advanced-http: an error occured during post processing response: Illegal break statement

code looks like this:

```
    const httpOptions = {
      method: 'post',
      data: mydataobject,
      serializer: 'json',
      timeout: 300,
      headers: { "Authorization": "Bearer " + pluginConstants.sync.token, "Content-type": "application/json", "Ocp-Apim-Subscription-Key": subscriptionkey }
    };

    cordova.plugin.http.sendRequest(url, httpOptions, function(response) {
      console.log("request success: " + response.status);
      console.log(JSON.stringify(response));
    }, function(response) {
      console.error("request error: " + response.error);
      console.log(JSON.stringify(response));
    });
```

response object in error callback looks like:

```
{
"status":-7,
"error":"advanced-http: an error occured during post processing response: Illegal break statement",
"url":"https://my-server-address.com/api",
"headers":{
  "date":"Tue, 22 Oct 2019 15:18:34 GMT",
  "content-length":"0",
  "x-android-selected-protocol":"http/1.1",
  "x-android-response-source":"NETWORK 200",
  "x-android-received-millis":"1571757513004",
  "access-control-allow-headers":"Authorization, Ocp-Apim-Subscription-Key, Content-Type, Accept",
  "x-android-sent-millis":"1571757512336",
  "x-frame-options":"SAMEORIGIN",
  "access-control-allow-methods":"GET, PUT, POST, DELETE, HEAD",
  "strict-transport-security":"max-age=31536000; includeSubdomains",
  "access-control-expose-headers":"*",
  "access-control-allow-origin":"*",
  "access-control-max-age":"3628800",
  "content-security-policy":"default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval';style-src 'self' 'unsafe-inline';font-src 'self';object-src 'none'",
  "x-content-type-options":"nosniff",
  "x-xss-protection":"1; mode=block",
  "referrer-policy":"no-referrer-when-downgrade",
  "connection":"keep-alive",
  "cache-control":"max-age=31536000; no-store"
  }
}
```

I've used chrome dev tools debugger, looks like injectRawResponseHandler in helper.js calls failure callback due to some exception. Debugger (breakpoint at line 317 with failure() call) shows that response object has status 200, so the request was successful, it's just post processing that fails.

Is it something I did wrong ? Or maybe a plugin bug ?

**Comment by silkimen** [comment]
Hi Piotr, did you also configure a responseType? Can you please set a breakpoint in line 288 and then log out response object (before it's changed to an error object).

**Comment by PiotrSmietana89** [comment]
Hi, sorry for late answer

Actually, I've managed to solve this issue, but since this happened kinda automagically when I've been applying changes to other parts of code (also HTTP related), I wasn't really able tell what exactly fixed this particular problem, then moved on with the project and forget about it.

The realization came today when I was reviewing some WIP commits in my git repo. I've noticed that, coincidentally, when I've made the other HTTP changes, I've also did a little code cleanup in code experiencing issue I've described in my original post. Some log changes, removing commented out lines, and removing break statement that was out of place in a callback. Yes, you read it correctly - I had a break statement in my cordova.plugin.http.sendRequest() success callback. It got there when I was copying code from previous implementation which, using XMLHttpRequest, was placed inside a switch statement, hence the break.

I'm new to Cordova and Javascript, and for most of the time I use Java. Looks like it was another "But why didn't compiler find thi... oh, wait" moment ;)

I've created a a simple custom-build localhost server (sending same set of headers and empty response body), and test app to reproduce this. Without break statement in success callback it works like a charm, but putting break into success callback reproduces the problem.

Such a stupid mistake. I feel ashamed I didn't notice that immediately.

I was led astray by a fact that I've never got success callback logs, even if break statement was in last line of success callback, after the logs. All I was getting were failure callback logs (messages starting with "advanced-http") with error suggesting there was something wrong "during post processing response" (it sounds like it's something that plugin does, not the callback code) and debugger pointed at catch block inside your plugin, hence my conviction that it might be a plugin bug. 

Sorry for all the ruckus. Since I've solved the issue and there's nothing wrong with the plugin, I'm closing this topic.
